pkg/occlient/occlient.go

```diff
@@ -793,3 +827,63 @@ func (c *Client) GetLabelValues(project string, label string, selector string) (
 
 	return values, nil
 }
+
+// GetClusterServiceClasses queries the service service catalog to get all the
+// currently available cluster service classes
+func (c *Client) GetClusterServiceClasses() ([]v1beta1.ClusterServiceClass, error) {
+	classList, err := c.serviceCatalogClient.ClusterServiceClasses().List(metav1.ListOptions{})
+	if err != nil {
+		return nil, errors.Wrap(err, "unable to list cluster service classes")
+	}
+	return classList.Items, nil
+}
+
+// GetClusterServiceClassExternalNames returns the names of all the cluster service
+// classes in the cluster
+func (c *Client) GetClusterServiceClassExternalNames() ([]string, error) {
+	var classNames []string
+
+	classes, err := c.GetClusterServiceClasses()
+	if err != nil {
+		return nil, errors.Wrap(err, "unable to get cluster service classes")
+	}
+
+	for _, class := range classes {
+		classNames = append(classNames, class.Spec.ExternalName)
+	}
+	return classNames, nil
+}
+
+// imageStreamExists returns true if the given image stream exists in the given
+// namespace
+func (c *Client) imageStreamExists(name string, namespace string) bool {
+	imageStreams, err := c.GetImageStreamsNames(namespace)
+	if err != nil {
+		log.Debugf("unable to get image streams in the namespace: %v", namespace)
+		return false
+	}
+
+	for _, is := range imageStreams {
+		if is == name {
+			return true
+		}
+	}
+	return false
+}
+
+// clusterServiceClassExists returns true if the given external name of the
+// cluster service class exists in the cluster, and false otherwise
+func (c *Client) clusterServiceClassExists(name string) bool {
```

Comment:
we're not using this in this PR, but we will need this when we implement deploying using s2i with this, so maybe we can keep this util function here